website/content/v1.10/kubernetes-guides/configuration/inlinemanifests.md

```diff
@@ -0,0 +1,149 @@
+
+Here’s an example of how to configure a cluster using an `inlineManifest`:
+
+```yaml
+cluster:
+  inlineManifests:
+    - name: my-app
+      contents: |
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+          name: my-application
+        spec:
+          # ... deployment specification
+```
+
+## extraManifests
+
+`extraManifests` are Kubernetes manifests fetched from external, unauthenticated HTTP sources such as GitHub, raw file servers, or gists.
+
+You define them in the `extraManifests` section of the machine configuration.
+They’re best suited for shared, versioned, or centrally managed resources.
+
+These manifests are pulled directly by the node during configuration.
+If the node doesn’t have network access to the HTTP endpoint hosting the manifest the installation will fail.
+
+Similarly, if the endpoint is down or returns an error, the manifest will not be applied, and the machine configuration will fail as a result.
+
+Here’s how to reference `extraManifests`:
+
+```yaml
+cluster:
+  extraManifests:
+    - "https://raw.githubusercontent.com/example/repo/main/manifest.yaml"
+    - "https://gist.githubusercontent.com/user/gist-id/raw/manifest.yaml"
+```
+
+## Resource Ordering Considerations
+
+Talos automatically sorts all manifests, including `inlineManifests`, `extraManifests`, or `built-in manifests`, before applying them in the following order:
+
+1. `Namespace` resources
+2. CustomResourceDefinitions (CRDs)
+3. All other resources, sorted alphabetically by their `metadata.name` property
+
+## Example Usecase: Install a GitOps controller with extraManifests
+
+A common use case for `inlineManifests` or `extraManifests` is to install a GitOps controller like Flux or ArgoCD.
+Once the controller is running, it connects to your Git repository and automatically applies the rest of your Kubernetes configuration.
+
+Here's how to install the Flux GitOps controller using an `extraManifest`:
+
+1. Create a patch file named `flux-extra-manifest.yaml` that automatically downloads and applies the Flux installation manifest from GitHub:
+
+    ```shell
+    cat << EOF > flux-extra-manifest.yaml
+    cluster:
+    extraManifests:
+        - "https://github.com/fluxcd/flux2/releases/latest/download/install.yaml"
+    EOF
+    ```
+
+1. Create a `CP_IPS` variable that contains the IP addresses of your control plane nodes:
+
+    ```bash
+    CP_IPS="<control-plane-ip-1>,<control-plane-ip-2>,<control-plane-ip-3>"
+    ```
+
+1. Run this command to export your `TALOSCONFIG` variable.
+You can skip this step if you've already done it:
+
+    ```bash
+    mkdir -p ~/.talos
+    cp ./talosconfig ~/.talos/config
+    ```
+
+1. Apply the `flux-extra-manifest.yaml` patch to your control plane nodes:
+
+    ```bash
+    talosctl patch machineconfig \
+    --patch @flux-extra-manifest.yaml \
+    --endpoints $CP_IPS \
+    --nodes $CP_IPS
+    ```
+
+1. Reboot the nodes.
```

Comment:
Can you confirm if a reboot is required for this patch? Because extra and inline manifests only apply during install or upgrade applying the patch and rebooting I don't think will trigger them to be installed if Talos is already installed and running.

Reply:
If you add a new manifest, not reboot is ever required, Talos will deploy it immediately.